### DISCUSSION TITLE
Cities start displaying at zoom level 7, even though my style says minzoom = 4

### DISCUSSION THREAD

**Discussion by nielsbom** [protomaps/basemaps#252] [Q&A] [answered]
Using the minimal style file below on [Maputnik](https://maplibre.org/maputnik/?). (I removed all of the other layers of my map)

As you can see the `places_city` layer is configured to show cities from zoomlevel 4, but for some reason cities only show up at zoomlevel 7 and above.

Why is this? I've seen PMTiles maps showing cities at higher zoomlevels, one example is the frontpage of https://docs.protomaps.com/

Thanks!

[empty_style.json](https://github.com/user-attachments/files/15515782/empty_style.json)

**Comment by nvkelso** [comment, marked as answer]
If you change your style layer filter they will show up earlier, as there are lower zoom features available in the places mvt data layer. 

Currently you're styling based on OSM `place=*` tags in the `kind_detail` Protomaps/Tilezen namespace which are only available at zoom 7+, but if you loosen the filter to be any `kind` = `locality` (which spans across city, town, and other values) then it'll work at low and high zooms.

For compatibility, probably all the lower zoom localities should have `city` backfilled into the `kind_detail`? Or whatever OSM has for that place versus Natural Earth? We do a partial data join now, we could expand that to include more properties.

But philosophically I would generally be trusting the tiles to show the right content (and we can modify them if there's a bug), and changing the icon and text styling based on another property – like the population rank (which has a higher fill rate and more consistent approach than OSM's place=* tag which varies widely between countries and even inside countries).

To: 

```
  "filter": [
    "all",
    ["==", "pmap:kind", "locality"]
  ],
```

From:

```
  "filter": [
    "all",
    ["==", "pmap:kind_detail", "city"]
  ],
```

**Comment by nielsbom** [comment]
Ah that makes sense, got it working now. Thank you!

Would it help if I open a PR against [the basemap docs](https://github.com/protomaps/docs/blob/main/basemaps/layers.md) to make this clear(er) to others? Or is it better to reference how OSM does it?

**Reply by nvkelso** [reply]
That's a good question, I don't have a ready answer so let's dive into it a bit more to see if something falls out?

Here's how Natural Earth, Who's On First, and Tilezen (which Protomaps has adopted) think about it:

pop_rank | min_pop | max_pop | colloquial | density (pop per sq km)
-- | -- | -- | -- | --
14 | 10,000,000 | 20,000,000 | megacity XL | 100,000
13 | 5,000,000 | 10,000,000 | megacity L | 75,000
12 | 1,000,000 | 5,000,000 | megacity M | 50,000
11 | 500,000 | 1,000,000 | megacity S (global city) | 20,000
10 | 200,000 | 500,000 | city XXXL | 15,000
9 | 100,000 | 200,000 | city XXL | 10,000
8 | 50,000 | 100,000 | city XL | 5,000
7 | 20,000 | 50,000 | city L | 2,500
6 | 10,000 | 20,000 | city M | 1,500
5 | 5,000 | 10,000 | city S | 1,000
4 | 2,000 | 5,000 | city XS (town L) | 800
3 | 1,000 | 2,000 | city XXS (town M) | 350
2 | 200 | 1,000 | village (town S) | 175
1 | 1 | 200 | hamlet | 750

In OSM there isn't a lot of standardization in hamlet, village, town, and city tags vis-a-vis population. But we could make a mapping like:

place= | pop_rank | min_pop | max_pop | colloquial
-- | -- | -- | -- | --
[city](https://wiki.openstreetmap.org/wiki/Tag:place%3Dcity) | 6-14 | 10,000 | 20,000,000 | city M - city XXXL, megacity S-XL
[town](https://wiki.openstreetmap.org/wiki/Tag:place%3Dtown) | 3, 4, 5 | 1,000 | 10,000 | city XXS (town M) - city S (town L)
[village](https://wiki.openstreetmap.org/wiki/Tag:place%3Dvillage) | 2 | 200 | 1,000 | village (town S)
[hamlet](https://wiki.openstreetmap.org/wiki/Tag:place%3Dhamlet) | 1 | 1 | 200 | hamlet

So if we wanted to backfill `kind_detail` from Natural Earth by population for the low zooms so the values were more or less consistent with higher zooms, we could use those population cutoffs with a code change (and maybe a documentation change).

But ideally the low zooms and high zooms have the same values for the same place. Else the style is going to stutter when zooming in and out, which we generally want to avoid. So we probably need to think more about the ad hoc data join and expand that for additional data properties (eg https://github.com/protomaps/basemaps/blob/main/tiles/src/main/java/com/protomaps/basemap/layers/Places.java#L199).

Either way, the `population_rank` is better to style on. If you want a very large city (like capitals of the provinces) then you could style rank 9+ a little larger size and bold. Keeping to the "rule of 3", you could also consolidate village and hamlet features into a single style class.

**Reply by nvkelso** [reply]
Looking more closely at https://github.com/protomaps/basemaps/blob/main/tiles/src/main/java/com/protomaps/basemap/layers/Places.java#L206-L230

Protomap's implementation of the Tilezen schema currently has the following population count (and imputed population rank) backfills per `place=*` value:

- `city` = 10,000
- `town` = 5,000
- `village` = 2,000
- `hamlet` isn't included (probably an error)
